### PR TITLE
Add copula-based TrueMeasure support

### PR DESCRIPTION
**Pull request by Laasya-73**
### Summary

This PR adds initial copula-based `TrueMeasure` support in QMCPy.

It introduces five copula classes:

- `GaussianCopula` for correlation-based dependence
- `StudentTCopula` for heavier joint-tail dependence
- `ClaytonCopula` for lower-tail dependence
- `GumbelCopula` for upper-tail dependence
- `FrankCopula` for symmetric Archimedean dependence

The common workflow is:

```text
Uniform QMC samples
-> copula transform to dependent uniforms
-> marginal inverse CDFs
-> final dependent target samples
```

#### Changes
- Added copula true measure classes
- Added validation for dimensions, parameters, marginals, and correlation matrices where applicable
- Added tests for shape, invalid inputs, finite outputs, dependence behavior, and parameter boundary cases
- Added a copula examples notebook
- Added the demo to the docs navigation through mkdocs.yml
- Updated doctest examples using the Examples: syntax

#### Notes
The implementation supports general dimensions where mathematically valid. For Frank copula, positive theta supports general dimension, while negative theta is restricted to the bivariate case.

### CODE REVIEW

**Comment by sou-cheng-choi** [comment]
* Update `true_measures.md `to document the new copula classes.
* Replace `assert` and bare `except Exception` to specific types (e.g., except (AttributeError, TypeError, ValueError).
* Resolve merge conflicts above

**Review by alegresor — CHANGES_REQUESTED** [review]
Really nice addition @Laasya-73! A few presentation suggestions are below. 

1. Expand doctests to print out samples, print the object itself, test for independent replications, and test any other settings of interest e.g. Frank copula with negative theta where you switch to the restriction to bivariate. The doctests should follow something like the [Gaussian class](https://github.com/QMCSoftware/QMCSoftware/blob/master/qmcpy/true_measure/gaussian.py)
2. Add references into the docs whose implementation you follow, e.g. https://arxiv.org/pdf/1508.03483 and/or others? 
3. In the future, please create a branch off of QMCSoftware rather than a fork, this makes it easier for us to checkout your code.

File: qmcpy/true_measure/gaussian_copula.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+               [ 0.6874,  4.955 ],
+               [ 0.2348,  5.3863]])
+        >>> tm  # doctest: +ELLIPSIS
+        GaussianCopula (AbstractTrueMeasure)
+            marginals       [<...rv_continuous_frozen object at ...>
+                             <...rv_continuous_frozen object at ...>]
+            correlation     [[1.  0.6]
+                             [0.6 1. ]]
+        >>> rep_tm = GaussianCopula(DigitalNetB2(2, seed=7, replications=2), marginals=marginals, correlation=corr)
+        >>> rep_tm(4).shape
```
Comment (alegresor) [inline qmcpy/true_measure/gaussian_copula.py]: Please print the samples themselves after printing the shape. I also recommend using `d=3` and `n=4` and `R=2` so the samples have shape `(2,4,3)` and it is clear which dimension corresponds to `R`, `n`, and `d`. The current setup may confuse readers between which `2` is for replications and which is for dimension. Please change this for the other copulas as well

**Review by alegresor — APPROVED** [review]
Things look fine from my end, modulo the one minor suggestion. Please wait for approval from @sou-cheng-choi and @fredhickernell before merging

**Review by sou-cheng-choi — CHANGES_REQUESTED** [review]
1. Please remove the figures folder and do not keep image files like .png
2. Suggest to have a class Copula in `copula.py` that is a child class of AbstractTrueMeasure and parent of other coupla classes. 
3. Typos in demo notebook:

    (a) change 'uniforms' to 'uniform' in "the samples shown here are still uniforms" and "The points are still uniforms"
    (b)  Change "focus dependence" to "focus on dependence"
    
4. Following our review meeting, we believe the following additions would significantly improve the notebook:

    (a) Include references (chapter, theorem, or page number) for the source material used in the book.
    (b) Add more detailed definitions and explanations for the notation, preferably maintaining consistency with the sources you have referenced.

**Comment by sou-cheng-choi** [comment]
@Laasya-73 

1. "make doc" results in a few warning/errors that need to be fixed in `mkdocs.yml`

```
WARNING -  A reference to 'demos/copula_examples/copula_examples.ipynb' is included in the 'nav' configuration, which is not
           found in the documentation files.
INFO    -  mkdocstrings_handlers: Formatting signatures requires either Black or Ruff to be installed.
ERROR   -  mkdocstrings: qmcpy.true_measure.copula.Copula could not be found
ERROR   -  Error reading page 'api/true_measures.md':
ERROR   -  Could not collect 'qmcpy.true_measure.copula.Copula'
```


2. Could we put `_unit_weight_with_warning` in the base Copula class and take out the duplicate definitions in the subclasses?

**Review by sou-cheng-choi — CHANGES_REQUESTED** [review]
See the comments above.

**Comment by fjhickernell** [comment]
Laasya has asked us all to review please.

**Comment by sou-cheng-choi** [comment]
@Laasya-73 

Thank you for your continuous improvement on this important feature. I have a few minor comments below:

1. In the demo's first code cell, we should be able to simply import qmcpy packages using `from qmcpy import DigitalNetB2, ClaytonCopula, FrankCopula, GaussianCopula, GumbelCopula, IIDStdUniform, StudentTCopula`. LIkewise, you could shorten all qmcpy related import statements in the `test_*.py`.

2. Merge `test_copula_branches.py` and `est_gaussian_copula.py` into one file such as `test_copulas.py`, but keep clear sections.

3. The line containing `paths` in `mkdocs.yml` should not be changed.

4. How is the package `ruff` in `project.toml` being used? Do we really need it?

5.  The documentation in browser showing the following literally for Copula. 

      .. math::
      
      ```
      U \in [0,1]^d \mapsto V = T(U) \in [0,1]^d.
      ```
    Please implement the conversion of all docstring math blocks from reST directives to standard Markdown/KaTeX format ($$ ... $$).

**Review by sou-cheng-choi — CHANGES_REQUESTED** [review]
Please see the comments above

**Comment by Laasya-73** [comment]
@sou-cheng-choi,

I have pushed the updates addressing the comments above. Could you please review the changes when you have a chance?
Thank You:))

**Review by alegresor — APPROVED** [review]
Looks great @Laasya-73! One minor suggestion is to rename `Copula` to `AbstractCopula` so users know not to instantiate it.

**Comment by alegresor** [comment]
@Laasya-73, please wait for @fredhickernell and @sou-cheng-choi to review before merging.

**Comment by sou-cheng-choi** [comment]
@Laasya-73 

I have approved this PR. 

Please also wait for review from @fjhickernell . Thank you.

**Review by sou-cheng-choi — APPROVED** [review]

**Comment by Laasya-73** [comment]
@fjhickernell I have pushed the latest updates. Could you please take a look when you get a chance? 
Thank you!

**Review by fjhickernell — APPROVED** [review]
I am satisfied.
